addons/web_editor/static/src/components/media_dialog/media_dialog.js

```diff
@@ -258,6 +258,9 @@ export class MediaDialog extends Component {
             } else {
                 this.selectedMedia[tabId] = this.selectedMedia[tabId].filter(m => m.id !== media.id);
             }
+        } else if (media && Object.keys(media).length === 0) {
```

Comment:
Why don't we put this condition above if and return after setting ```selectedMedia``` if there is no media